### PR TITLE
refactor: refactor collect data modal for duplication case

### PR DESCRIPTION
**Pull request by skdud4659**
### To Reviewers
- [ ] Skip (`style`, `chore`, `ci` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [x] Bug fixes
- [ ] Feature improvement
- [x] Refactor
- [ ] Others (performance improvement, CI/CD, etc.)

### Affects to
- [ ] Packages
  - [ ] core-lib
  - [ ] mirinae
  - [ ] etc 

- [ ] Apps
  - [ ] storybook
  - [x] web

### Checklist
- Did you check the `lint` and `type`?
- Did you document the changes?
  - Changes in `mirinae` should be reflected in `storybook`.
- Did you test the app after the package changes?

### Description
SSIA
- There are no duplicate tasks.
  - Entire Accounts
image

  - Single Account
image

- There are duplicate tasks.
  - Entire Accounts
image

  - Single Account
image

### Things to Talk About
Sorry, there are quite a few files!
Plz check by commits

### CODE REVIEW

File: apps/web/src/services/asset-inventory/collector/shared/collector-data-modal/CollectorDataModal.vue [hunk 1]
```diff
@@ -52,38 +55,30 @@ import CollectorDataDefaultInner
...
+        if (!recentJob) return '';
+        return recentJob.status === JOB_STATE.IN_PROGRESS;
     }),
-    collectDataType: computed(() => {
-        const selectedSecret = collectorDataModalState.selectedSecret;
-        if (!selectedSecret) return COLLECT_DATA_TYPE.COLLECTOR;
-        return Object.keys(selectedSecret).length > 0 ? COLLECT_DATA_TYPE.SECRET : COLLECT_DATA_TYPE.COLLECTOR;
+    provider: computed(() => {
+        const selectedCollector = collectorDataModalState.selectedCollector;
+        return storeState.providers[selectedCollector?.provider || ''];
```
Comment (yuda110) [inline apps/web/src/services/asset-inventory/collector/shared/collector-data-modal/CollectorDataModal.vue:80]: suggestion: Searching for empty strings looks a bit odd, so this seems better!

```typescript
return selectedCollector?.provider ? storeState.providers[selectedCollector.provider] : undefined;
```

**Review by yuda110 — APPROVED** [review]
LGTM!!!

**Review by WANZARGEN — APPROVED** [review]
LGTM overally~ I'd like to recommend u to declare types for each states. It would be much helpful for reducing reviewing time 😄

File: apps/web/src/services/asset-inventory/collector/shared/collector-data-modal/CollectorDataModal.vue [hunk 2]
```diff
@@ -112,4 +107,38 @@ const handleClickConfirm = async () => {
...
+        ErrorHandler.handleError(e);
+        collectorDataModalStore.$patch((_state) => {
+            _state.secrets = [];
+        });
+    }
+};
+
+watch([() => collectorDataModalState.selectedCollector, () => collectorDataModalState.visible], async ([selectedCollector, visible]) => {
+    if (!selectedCollector || !visible) return;
+    await fetchSecrets(selectedCollector.provider);
```
[the remark this answers is not in the document]
Comment (skdud4659) [reply apps/web/src/services/asset-inventory/collector/shared/collector-data-modal/CollectorDataModal.vue:133]: At the store, when clicking on an individual account, we store information about the selected secrets. However, in the modal that appears when there are no duplicate tasks, we needed to fetch the total count of secrets separately to display.

File: apps/web/src/services/asset-inventory/collector/shared/collector-data-modal/modules/CollectorDataDefaultInner.vue [hunk 3]
```diff
@@ -2,40 +2,55 @@
...
 });
 
+const state = reactive({
+    serviceAccountName: computed(() => {
+        const selectedSecret = collectorDataModalState.selectedSecret;
+        if (!selectedSecret) return '';
+
+        const id = selectedSecret.service_account_id;
+        const fullName = selectedSecret.name;
+        return fullName.split(id)[0];
```
[the remark this answers is not in the document]
Comment (skdud4659) [reply apps/web/src/services/asset-inventory/collector/shared/collector-data-modal/modules/CollectorDataDefaultInner.vue:51]: I noticed that the 'selectedSecret' stores the name information along with an accompanying ID, which should not be exposed earlier. To display only the pure name, we created this logic.

**Pull request merged**